go-controller/pkg/ovn/ovn.go

```diff
@@ -549,9 +561,62 @@ func (oc *Controller) recordPodEvent(addErr error, pod *kapi.Pod) {
 	}
 }
 
+// iterateRetryPods checks if any outstanding pods have been waiting for 60 seconds of last known failure
+// then tries to re-add them if so
+func (oc *Controller) iterateRetryPods() {
+	oc.retryPodsLock.Lock()
+	defer oc.retryPodsLock.Unlock()
+	now := time.Now()
+	for uid, podEntry := range oc.retryPods {
+		pod := podEntry.pod
+		podTimer := podEntry.timeStamp.Add(time.Minute)
+		if podScheduled(pod) && util.PodWantsNetwork(pod) && now.After(podTimer) {
+			klog.Infof("Pod Retry Loop: adding pod %s", pod.Name)
+			if err := oc.addLogicalPort(pod); err != nil {
```

Comment:
Is it really safe to call `addLogicalPort` from a random thread? Currently we only ever call it from the `WatchPods` thread, so it doesn't have to worry about locking...

Reply:
I tried to ensure addLogicalPort could not be called for the same pod at the same time with the current locking of the map. 
 1. Add doesn't care because the pod cant exist in the map:
 2. Update waits until it can get the lock and removes the pod from map to make sure the goroutine wont add it.
 3. Delete waits until lock and removes.

This way the controller add/update wont block each other during add logical port, but the goroutine will block the update/delete operations every 30 seconds while it finishes all its add logical ports.

Reply:
I was worried about locking much more generically than that... in the past, ovn-kubernetes was terrible about locking, and any time you tried to move anything to a new thread, stuff would blow up. Maybe that's better these days...